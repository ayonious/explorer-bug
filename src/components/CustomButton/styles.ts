import styled from 'styled-components';

export const CustomButtonDiv = styled.div`
  margin: 10px;
`;
