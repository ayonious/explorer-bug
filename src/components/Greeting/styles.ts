import styled from 'styled-components';

export const CustomDiv = styled.div`
  display: flex;
  align-content: center;
  justify-content: center;
`;
