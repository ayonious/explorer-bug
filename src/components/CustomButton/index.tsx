import Button from '@mui/material/Button';
import * as React from 'react';

import { CustomButtonDiv } from './styles';

const CustomButton = (props: any) => {
  return (
    <CustomButtonDiv>
      <Button variant="contained" color="primary" onClick={props.onClick}>
        {props.title}
      </Button>
    </CustomButtonDiv>
  );
};

export default CustomButton;
