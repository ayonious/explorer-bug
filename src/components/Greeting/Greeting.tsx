import * as React from 'react';

import { CustomDiv } from './styles';

const Greeting = () => {
  return (
    <CustomDiv>
      <h1> Move the bug to eat all food! </h1>
    </CustomDiv>
  );
};

export default Greeting;
