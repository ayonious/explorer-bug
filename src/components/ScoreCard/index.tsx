import * as React from 'react';

import { CustomDiv } from './styles';

const ScoreCard = ({ score }: { score: number }) => {
  return (
    <CustomDiv>
      <h1> Score: {score} </h1>
    </CustomDiv>
  );
};

export default ScoreCard;
