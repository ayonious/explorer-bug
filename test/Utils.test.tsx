import * as React from 'react';
import { render } from '@testing-library/react';
import CustomButton from '../src/components/CustomButton';
import ScoreCard from '../src/components/ScoreCard';
import Greeting from '../src/components/Greeting';

it('ScoreCard renders', () => {
  const tree = render(<ScoreCard score={10} />);
  expect(tree).toMatchSnapshot();
});

it('Greeting renders', () => {
  const tree = render(<Greeting />);
  expect(tree).toMatchSnapshot();
});

it('CustomButton renders', () => {
  const tree = render(<CustomButton title={'Score of Game'} />);
  expect(tree).toMatchSnapshot();
});
