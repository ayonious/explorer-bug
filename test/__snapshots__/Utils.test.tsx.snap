// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomButton renders 1`] = `
<div
  className="sc-AykKC cDtWnO"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Score of Game
    </span>
  </button>
</div>
`;

exports[`Greeting renders 1`] = `
<div
  className="sc-AykKD lfrAza"
>
  <h1>
     Move the bug to eat all food! 
  </h1>
</div>
`;

exports[`ScoreCard renders 1`] = `
<div
  className="sc-AykKD lfrAza"
>
  <h1>
     Score: 
    10
     
  </h1>
</div>
`;
